backend/siarnaq/bracket/challonge.py

```diff
@@ -0,0 +1,269 @@
+                    raise RuntimeError(
+                        "The bracket service's round does not only\
+                            have matches that are ready."
+                    )
+                challonge_matches.append(item)
+
+            case {"type": "participant", "id": id}:
+                challonge_participants[id] = item
+
+    match_objects = []
+    match_participant_objects = []
+
+    for challonge_match in challonge_matches:
+        match_object = apps.get_model("compete", "Match")(
+            episode=round.tournament.episode,
+            tournament_round=round,
+            alternate_order=True,
+            is_ranked=False,
+            challonge_id=challonge_match["id"],
+        )
+        match_objects.append(match_object)
+
+        # Note that Challonge 1-indexes its player indexes
+        # while Saturn 0-indexes.
+        for (challonge_player_index, saturn_player_index) in (
+            ("player1", 0),
+            ("player2", 1),
+        ):
+            # This looks ugly but it's how to parse through the Challonge-related data.
+            challonge_participant_id = challonge_match["relationships"][
+                challonge_player_index
+            ]["data"]["id"]
+            misc_key = json.loads(
+                challonge_participants[challonge_participant_id]["attributes"]["misc"]
+            )
+            team_id = misc_key["team_id"]
+            submission_id = misc_key["submission_id"]
+
+            match_participant_object = apps.get_model("compete", "MatchParticipant")(
+                team_id=team_id,
+                submission_id=submission_id,
+                match=match_object,
+                player_index=saturn_player_index,
+                challonge_id=challonge_participant_id,
+            )
+            match_participant_objects.append(match_participant_object)
+
+    return match_objects, match_participant_objects
+
+
+def update_match(match: Match, is_private):
+    tournament: Tournament = match.tournament_round.tournament
+
+    tournament_challonge_id = (
+        tournament.bracket_id_private if is_private else tournament.bracket_id_public
+    )
+
+    match_challonge_id = match.bracket_id_private if is_private else None
+
+    url = f"{URL_BASE}tournaments/{tournament_challonge_id}/\
+        matches/{match_challonge_id}.json"
+
+    # Wrangle the Match object into a format Challonge likes.
+    # In particular, Challonge wants an array of objects,
+    # each of which represents a participant's data.
+    # The data is id, score, and whether or not they advance.
+
+    # To assign the "advance" flag, compute the high score,
+    # and then set the flag for those that have the high score.
+
+    # Also, we convert "score" to a string for Challonge.
+    # (This is required because Challonge supports scores of sets,
+    # which are comma-delimited lists of numbers.
+    # We don't use this though)
+
+    participants = match.participants.all()
```

Comment:
Convert it to a list here, so that it doesn't generate a query for each time you iterate.

```suggestion
    participants = list(match.participants.all())
```